Backendless/Push/Registrar.cs

```diff
@@ -1,49 +1,81 @@
 ﻿using System;
-using BackendlessAPI.Async;
-using BackendlessAPI.Exception;
-#if WINDOWS_PHONE || WINDOWS_PHONE8
-using Microsoft.Phone.Notification;
+using System.Collections.Generic;
+using BackendlessAPI.Messaging;
+using BackendlessAPI.Service;
+using BackendlessAPI.Engine;
 
 namespace BackendlessAPI.Push
 {
   internal static class Registrar
   {
-    private static RegistrationDecorator _currentRegistration;
+    internal static String RegisterDevice( String token, List<String> channels, DateTime? expiration )
+    {
+      DeviceRegistration deviceRegistration = new DeviceRegistration();
+      deviceRegistration.Channels = channels;
+      deviceRegistration.Expiration = expiration;
+      deviceRegistration.DeviceToken = token;
+      deviceRegistration.DeviceId = Backendless.Messaging.DeviceID;
+      deviceRegistration.Channels = channels;
+      
+      if( Environment.OSVersion.Platform.ToString() == "Unix" )

```

Comment:
@holailusoria this does not look right. Is there a better way to detect that the current environment is Android? For example https://stackoverflow.com/a/44653014

Reply:
@markpiller  this can only be used in Xamarin Project. But .NET-SDK is not a Xamarin project

Reply:
@markpiller This is the best way I have found